src/System.Drawing.Design.Primitives/src/System/Drawing/Design/ToolboxItem.cs

```diff
@@ -0,0 +1,1100 @@
+            IComponent[] comps = CreateComponentsCore(host, new Hashtable());
+            if (comps != null && comps.Length > 0)
+            {
+                OnComponentsCreated(new ToolboxComponentsCreatedEventArgs(comps));
+            }
+            return comps;
+        }
+
+        /// <summary>
+        /// Creates objects from the type contained in this toolbox item.  If designerHost is non-null
+        /// this will also add them to the designer.
+        /// </summary>
+        /// <returns></returns>
+        public IComponent[] CreateComponents(IDesignerHost host, IDictionary defaultValues)
+        {
+            OnComponentsCreating(new ToolboxComponentsCreatingEventArgs(host));
+            IComponent[] comps = CreateComponentsCore(host, defaultValues);
+            if (comps != null && comps.Length > 0)
+            {
+                OnComponentsCreated(new ToolboxComponentsCreatedEventArgs(comps));
+            }
+            return comps;
+        }
+
+        /// <summary>
+        /// Creates objects from the type contained in this toolbox item.  If designerHost is non-null
+        /// this will also add them to the designer.
+        /// </summary>
+        protected virtual IComponent[] CreateComponentsCore(IDesignerHost host)
+        {
+            ArrayList comps = new ArrayList();
+
+            Type createType = GetType(host, AssemblyName, TypeName, true);
+            if (createType != null)
+            {
+                if (host != null)
+                {
+                    comps.Add(host.CreateComponent(createType));
+                }
+                else if (typeof(IComponent).IsAssignableFrom(createType))
+                {
+                    comps.Add(TypeDescriptor.CreateInstance(null, createType, null, null));
+                }
+            }
+
+            IComponent[] temp = new IComponent[comps.Count];
+            comps.CopyTo(temp, 0);
+            return temp;
+        }
+
+        /// <summary>
+        /// Creates objects from the type contained in this toolbox item.  If designerHost is non-null
+        ///     this will also add them to the designer.
+        /// </summary>
+        protected virtual IComponent[] CreateComponentsCore(IDesignerHost host, IDictionary defaultValues)
+        {
+            IComponent[] components = CreateComponentsCore(host);
+
+            if (host != null)
+            {
+                for (int i = 0; i < components.Length; i++)
+                {
+                    IComponentInitializer init = host.GetDesigner(components[i]) as IComponentInitializer;
+                    if (init != null)
+                    {
+                        bool removeComponent = true;
+
+                        try
+                        {
+                            init.InitializeNewComponent(defaultValues);
+                            removeComponent = false;
+                        }
+                        finally
```

Comment:
I don't understand this pattern. Why isn't this just a catch without the boolean variable? The finally block is only executed if the try block run successfully.

Reply:
It was ported as-is from desktop. Rearranging it without using binary. No specific exception handling though.